### PR TITLE
feat: tab component

### PR DESCRIPTION
**Pull request by Damien-MATHIEU**
## Description
Add Tab component https://www.figma.com/file/nPJGejQqgLQfkVMJqhcCeO/Maquettes?node-id=38%3A288

## Changes
Add Tag folder in components

## Linked issues
#19

### CODE REVIEW

File: src/components/Tab/core/tab.module.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+  }
+
+  &--grey-light {
+    border-bottom: 2px map.get(colors.$colors, 'grey-light') solid;
+    color: map.get(colors.$colors, 'grey-light');
+
+    // &:hover {
+    //   background-color: map.get(colors.$colors, 'grey-light');
+    // }
```
Comment (nartydev) [inline src/components/Tab/core/tab.module.scss:41]: If we don't need hover on this color you can delete this part

File: src/components/Tab/index.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { FC } from 'react'
+
+import { createIcon, getClassNames } from './core/tab.helpers'
+import { IProps } from './core/tab.typings'
+
+export const Tab: FC<IProps> = ({
+  text,
+  color = 'grey-light',
```
Comment (nartydev) [inline src/components/Tab/index.tsx:8]: I think you can change `color` by `selected` for put more logic in component and less somewhere else

File: src/components/Tab/index.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  iconStart,
+  link,
+  className,
+}) => {
+  const classNames = getClassNames({ color, className })
+
+  return (
+    <a href={link} className={classNames.join(' ')}>
```
Comment (nartydev) [inline src/components/Tab/index.tsx:16]: Use `NavLink` instead of `<a>` from react-router-dom library

**Review by nartydev — DISMISSED** [review]

File: src/components/Tab/index.tsx (hunk 2 above)
Comment (Damien-MATHIEU) [reply src/components/Tab/index.tsx:8]: Do you think about `selected` being a boolean ? Because we have 3 colors on the figma

Comment (nartydev) [reply src/components/Tab/index.tsx:8]: Yes we can play with another props like "disabled" for light grey for example

Comment (Damien-MATHIEU) [reply src/components/Tab/index.tsx:8]: Alright

**Review by moreirathomas — COMMENTED** [review]
Nice!

Can we have an integration in `UI` page similarly the to others component?

I think the idea behind the ui page is very nice and adding every components from our ui kit to it seems the way to go :)

**Comment by Damien-MATHIEU** [comment]
Yes of course!

File: src/pages/UI/index.tsx [hunk 4]
```diff
@@ -1,7 +1,14 @@
...
+import {
+  CheckCircle,
+  Loader,
+  PieChart,
+  PlusSquare,
+  Settings,
+} from 'react-feather'
 
 import { Button, Typography } from '@/components'
+import { Tab } from '@/components/Tab'
```
Comment (moreirathomas) [inline src/pages/UI/index.tsx:11]: I suggest we follow the convention of importing from `@/components` here too.

In `@/components/index.ts`:

```ts
export { Tab } from './Tab'
```

```suggestion
import { Button, Tab, Typography } from '@/components'
```

File: src/components/Tab/index.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+}) => {
+  const classNames = getClassNames({ selected, disabled, className })
+
+  return (
+    <NavLink to={link} className={classNames.join(' ')}>
+      {iconStart && createIcon({ icon: iconStart })}
+      {text}
+    </NavLink>
+  )
+}
```
Comment (moreirathomas) [inline src/components/Tab/index.tsx:23]: How does the reactivity around the selected/active state is handled?

As we use a router to navigate from configure to run and to result sub views, the tabs here are use to navigate on those routes.

By passing `selected` props to display the selected/active state, I don't know how we're gonna easily know which on should be active.

Could we use `isActive` prop from `NavLink` in combination with the route passed to `to={link}`.

If we need finer handle on the state, we could search for react-router-dom [api](https://github.com/remix-run/react-router/tree/main/docs). I suspect `useRouteMatch` to be a good candidate.

Sidenote: @alex-mngt I see that react-router-dom version is pinned to exactly `6`, is there a reason for it?

**Review by moreirathomas — COMMENTED** [review]
Maybe we can also use the tabs inside `@/pages/Run` as it already use a navigation via router and `NavLink`.

```jsx
// src/pages/Run/index.tsx

<Tab link="./config" text="Configure" iconStart={Settings} />
<Tab link="./result" text="Results" iconStart={CheckCircle} />
```

By doing this we incrementally use our new components in the final product :slightly_smiling_face:

File: src/components/Tab/index.tsx (hunk 5 above)
Comment (alex-mngt) [reply src/components/Tab/index.tsx:23]: I do agree with @moreirathomas, it seems that using IsActive prop is much mure appropriate and as I remember it, the granularity given by this prop should do the job.

There is also no reason to keep react router pinned to version 6, I don't know how it ended pinned to an exact version ^^

**Comment by Damien-MATHIEU** [comment]
> Maybe we can also use the tabs inside `@/pages/Run` as it already use a navigation via router and `NavLink`.
> 
> ```js
> // src/pages/Run/index.tsx
> 
> <Tab link="./config" text="Configure" iconStart={Settings} />
> <Tab link="./result" text="Results" iconStart={CheckCircle} />
> ```
> 
> By doing this we incrementally use our new components in the final product 🙂

In the branch `feat/tab-component` i don't have the `Run` directory yet. Is there a way to get the commits to `main` to my branch, or do i need to commit this one and then do another PR for the changes in the `Run` directory?

**Comment by moreirathomas** [comment]
> In the branch `feat/tab-component` i don't have the `Run` directory yet. Is there a way to get the commits to `main` to my branch, or do i need to commit this one and then do another PR for the changes in the `Run` directory?

Simply merge `main` into `feat/tab-component`.

Or if you want to do it more cleanly you can try rebasing `main` into `feat/tab-component` then force push the feature branch.

In your case I'd go with a simple merge.

```sh
git pull origin main # update main branch locally
git checkout feat/tab-component # make sure you're on the feature branch, can be skipped
git merge main # merge progress from main into feature branch
```

**Review by alex-mngt — APPROVED** [review]

**Pull request merged**